e2e/src/test/scala/scalapb/zio_grpc/CallDriverSpec.scala

```diff
@@ -0,0 +1,28 @@
+package scalapb.zio_grpc
+
+import io.grpc.Status
+import scalapb.zio_grpc.TestUtils._
+import scalapb.zio_grpc.server.CallDriver
+import zio._
+import zio.duration._
+import zio.test.{DefaultRunnableSpec, _}
+
+object CallDriverSpec extends DefaultRunnableSpec {
+
+  def spec = suite("CallDriverSpec")(
+    testM("exitToStatus prioritizes failures over interrupts") {
+      val effectWithInterruptAndFailure = ZIO
+        .foreachParN_(128)(List.range(0, 16))(i =>
+          for {
+            delay <- random.nextIntBetween(100, 200)
+            _     <- environment.live(clock.sleep(delay.milliseconds))
```

Comment:
Can the logic be tested without a live sleep?

Reply:
I am not sure how to create an effect that would fail with a cause that contains both an interrupt and a failure without such a sleep.

It would be possible to make an artificial exit instead of the effect:
```scala
Exit.Failure(Cause.Both(Cause.Interrupt(fiberId), Cause.fail("failure")))
```